pyat/at/tracking/particles.py

```diff
@@ -0,0 +1,131 @@
+"""
+Functions relating to particle generation
+"""
+import numpy
+from at.physics import ohmi_envelope
+
+__all__ = ['beam', 'sigma_matrix']
+
+
+def sigma_matrix(argin, twiss=False):
+    """
+    Calculate the correlation matrix to be used for particle generation
+
+    PARAMETERS
+        argin           Lattice object or list of twiss parameters.
+                        Depending on the length of the list the function
+                        will do different things:
+                        [espread, blength] 
+                            computes 2x2 longitudinal sigma matrix
+                        [beta, alpha, emit] 
+                            computes 2x2 transverse sigma matrix
+                        [betax, alphax, emitx, betay, alphay, emity] 
+                            computes the 4x4 sigma matrix for horizontal 
+                            and vertical.
+                        [betax, alphax, emitx, betay, alphay, emity, espread, 
+                            blength] 
+                            computes the 6x6 sigma matrix.
+
+    KEYWORDS
+        twiss=False     Flag which states whether the input is a
+                        lattice object or list of twiss parameters
+
+    OUTPUT
+        sigma_matrix    correlation matrix (either 2x2, 4x4 or 6x6)
+
+    """
+    if not twiss:
+        cflag = False
+        if not argin.radiation:
+            argin.radiation_on()
+            cflag = True
```

Comment:
Rather that modifying in-place and reverting later, you could use:
`argin=argin.radiation_on(copy=True)`
to work on a local copy. No need to turn radiation off later

Reply:
Agreed. Much better solution.